### DISCUSSION TITLE
Plugin Registration

### DISCUSSION THREAD

**Discussion by asifjunaid** [MscrmTools/XrmToolBox#1058] [Q&A] [answered]
Hi,
I have registered a tool on xrmtoolbox which is still in pending status. How much time it takes. How do I know that if there is any issue on tool

**Comment by BetimBeja** [comment]
Hello @asifjunaid , tools registered for the first time are manually tested by @MscrmTools and it might take a while to receive a response. Anyways, I downloaded you NuGet package and it is not formatted correctly, so it will be rejected :( 
Please read here https://www.xrmtoolbox.com/documentation/for-developers/deploy-your-plugin-in-plugins-store/
What I noticed was the missing Plugins folder in your NuGet package. I didn't test it further.

**Reply by BetimBeja** [reply]
When I try opening you plugin I get a NullReferenceException
![image](https://user-images.githubusercontent.com/11160171/128601227-4bbb78b2-85d8-451b-a18a-acedb2a3fbd3.png)

**Reply by BetimBeja** [reply]
Have you tried your tool on XrmToolBox? I downloaded your latest source code, and tried to debug it with this configuration
![image](https://user-images.githubusercontent.com/11160171/128601635-95acd329-c134-429a-909a-1c4c163531f0.png)
But it doesn't start anyway

**Reply by BetimBeja** [reply]
After changing a little bit of your build process to make it debuggable (Why do you use ILMerge?) 
![image](https://user-images.githubusercontent.com/11160171/128602333-d5343612-9a8f-47f3-b474-f2e13e730c2a.png)
OrganizationServiceProxy is null. I use ClientSecret authentication by the way :)

**Reply by asifjunaid** [reply]
Hi @BetimBeja, thanks for looking the tool. I have addressed and fixed the issues related to ClientSecret authentication. I have added Plugins folder. I am using ILMerge to include Nuget.Core dll in plugin. In order to extract crm solution, I am downloading solution packager at runtime. On non development machine nuget.core dll is not available in GAC.

Please review and add the plugin :-)

**Reply by BetimBeja** [reply]
NuGet.Core is part of the XrmToolBox shipped assemblies. You don't have to include it in your tool. XrmToolBoxPackage v1.2021.5.47 has a dependency problem which should be fixed in the latest releases of XrmToolBox (@MscrmTools can you align XrmToolBoxPackage to the latest version of XrmToolBox?)

**Reply by asifjunaid** [reply]
@BetimBeja Nuget.Core is not available in  downloaded version of 1.2021.6.51. During my testing on non dev machine( by placing my dll on plugins folder )it was throwing nuget.core missing error that's why I use ILMerge.

**Reply by BetimBeja** [reply]
I stand corrected, NuGet.Core was the old dll that used the NuGet v2 endpoint which was deprecated. 
![image](https://user-images.githubusercontent.com/11160171/128671155-2a586cf2-f40e-4257-ae51-d47d70da25ed.png)
It should not be part of XrmToolBox anymore. If you need to consume NuGet packages you can take a look at my two tools or the code of XrmToolBox which have been both updated to use the latest NuGet v3 endpoint.

**Reply by asifjunaid** [reply]
Hi @BetimBeja, I have removed Nuget.Core dependency. Can you review and validate. Thanks

**Reply by MscrmTools** [reply]
There is a bug in the template which makes Nuget.Protocol is missing. Please add it to your project but do not ilmerge it.

As for tool validation, I'm on holidays so I cannot validate anything before two weeks maximum.
Maybe @daryllabar or @rappen can process with the validation. They have the rights to do it

**Reply by asifjunaid** [reply]
Thanks @MscrmTools, Sorry for bothering you during your holidays. Enjoy your Holidays :-) 

@daryllabar @rappen can you please check and validate. Nuget.Protocol is already added in my project as reference.

**Reply by rappen** [reply]
Hi @asifjunaid - what is your name on the XrmToolBox site, or the NuGet name / id? So I can find your tool...

**Reply by asifjunaid** [reply]
Hi @rappen, Name on XrmTooBox site is  "Muhammad Asif" and Tool Name is "CRM Solution Comparer" Nuget package Id is CRMSolutionComparer

**Reply by rappen** [reply]
I think this Plugin folder should be under `lib\net462` instead of the root...

![image](https://user-images.githubusercontent.com/2572253/128862900-acbd36d9-a9c7-41a5-8ab0-6ad54ecc830c.png)

Maybe @MscrmTools thinks this work now, but I don't think so.

**Reply by asifjunaid** [reply]
@rappen  you are right. I have moved it inside net462

**Reply by rappen** [reply]
The url for this is 404:
![image](https://user-images.githubusercontent.com/2572253/128914847-405c4a80-9283-4f50-83fb-e4f935d8ff0f.png)
Tried something: https://github.com/asif.junaid@hotmail.com/https://github.com/asifjunaid/SolutionComparer/issues/new?body=....

It is connect to different orgs, but thy show the same in Source and Target. How?

![image](https://user-images.githubusercontent.com/2572253/128915455-7d3a8faa-77d1-4d2b-9095-0e98aa72708b.png)
Is that right?

Unsure if it is ok, Connected to a new org for this tool (or all tools), it does not change it...

**Reply by asifjunaid** [reply]
@rappen  1. I have fixed the Feedback url issue.
2.  If user will click on "Connect To Target" it will connect to new organization for the current tool only. it is user choice that he can connect to same organization using target button. XRMToolBox connect button will connect to source only.
I also formatted "Target Organization Connected To" text
For target button I used XrmToolBox "AdditionalConnectionDetails" feature
when user will connect using Target button then XRMToolBox status bar connection detail will not change

**Comment by rappen** [comment, marked as answer]
👍
My stupid, had a bad connecting... 🙄
I think it looks good!

**Reply by asifjunaid** [reply]
Great @rappen It's cool. Thanks for your support.

Thanks @BetimBeja @MscrmTools. It is a wonderful community. All of your response time is meritorious.
